### PR TITLE
Clean up of devModeOptions

### PR DESCRIPTION
**Pull request by shamsasari**

### CODE REVIEW

**Comment by shamsasari** [comment]
Scaling back my changes. Making `testConfiguration` return NodeConfigurationImpl is not as simple as initially thought.

File: .idea/runConfigurations/BankOfCordaDriverKt___Issue_Web.xml [hunk 1]
```diff
@@ -1,6 +1,7 @@
 <component name="ProjectRunConfigurationManager">
   <configuration default="false" name="BankOfCordaDriverKt - Issue Web" type="JetRunConfigurationType" factoryName="Kotlin">
     <extension name="coverage" enabled="false" merge="false" sample_coverage="true" runner="idea" />
+    <extension name="krasa.grepconsole.plugin.runConfiguration.GrepRunConfigurationExtension" selectedProfileId="0" />
```
Comment (blsemo) [inline .idea/runConfigurations/BankOfCordaDriverKt___Issue_Web.xml]: Is adding this extension part of the clean-up? Is that used in IntelliJ or in some output?

File: node/src/main/kotlin/net/corda/node/services/config/NodeConfiguration.kt [hunk 2]
```diff
@@ -101,6 +97,8 @@ data class NodeConfigurationImpl(
         override val minimumPlatformVersion: Int = 1,
         override val rpcUsers: List<User>,
         override val verifierType: VerifierType,
+        // TODO typesafe config supports the notion of durations. Make use of that by mapping it to java.time.Duration.
```
Comment (blsemo) [inline node/src/main/kotlin/net/corda/node/services/config/NodeConfiguration.kt]: Shouldn't these TODO really be on the interface? Would be nice to have all proposed changes in one place, but not really important...

File: .idea/runConfigurations/BankOfCordaDriverKt___Issue_Web.xml (hunk 1 above)
Comment (shamsasari) [reply .idea/runConfigurations/BankOfCordaDriverKt___Issue_Web.xml]: No, its just that I have this plugin installed and Intellij insists on updating this file. This is going to be an ongoing issue I think.

File: node/src/main/kotlin/net/corda/node/services/config/NodeConfiguration.kt (hunk 2 above)
Comment (shamsasari) [reply node/src/main/kotlin/net/corda/node/services/config/NodeConfiguration.kt]: Probably, but I'm not convinced yeah need the separate interface...

**Pull request merged**